apps/next-evans/pages/[friend]/[friend_slug].tsx

```diff
@@ -0,0 +1,215 @@
+import cx from 'classnames';
+import { t } from '@friends-library/locale';
+import type { GetStaticPaths, GetStaticProps } from 'next';
+import type { Edition } from '@/lib/editions';
+import { LANG } from '@/lib/env';
+import { mostModernEdition } from '@/lib/editions';
+import FriendBlock from '@/components/pages/friend/FriendBlock';
+import FeaturedQuoteBlock from '@/components/pages/friend/FeaturedQuoteBlock';
+import BookByFriend from '@/components/pages/friend/BookByFriend';
+import TestimonialsBlock from '@/components/pages/friend/TestimonialsBlock';
+
+const client = new PrismaClient();
+
+export const getStaticPaths: GetStaticPaths = async () => {
+  const allFriends = await client.friends.findMany({
+    where: { lang: LANG },
+    select: { slug: true, gender: true },
+  });
+
+  const paths = allFriends.map((friend) => {
+    if (LANG === `en`) {
+      return {
+        params: { friend: `friend`, friend_slug: friend.slug },
+      };
+    }
+    return {
+      params: {
+        friend: friend.gender === `female` ? `amiga` : `amigo`,
+        friend_slug: friend.slug,
+      },
+    };
+  });
+
+  return { paths, fallback: false };
+};
+
+export const getStaticProps: GetStaticProps<Props> = async (context) => {
+  invariant(typeof context.params?.friend_slug === `string`);
+  const friend = await client.friends.findFirst({
+    where: {
+      slug: context.params.friend_slug,
+      lang: LANG,
+    },
+    select: {
+      name: true,
+      gender: true,
+      slug: true,
+      description: true,
+      friend_quotes: {
+        select: {
+          text: true,
+          source: true,
+        },
+      },
+      documents: {
+        select: {
+          title: true,
+          slug: true,
+          partial_description: true,
+          id: true,
+          document_tags: {
+            select: {
+              type: true,
+            },
+          },
+          editions: {
+            select: {
+              type: true,
+              edition_audios: {
+                select: {
+                  id: true,
+                },
+              },
+              isbns: {
+                select: {
+                  code: true,
+                },
+              },
+              downloads: {
+                select: {
+                  id: true,
+                },
+              },
+              edition_impressions: {
+                select: {
+                  paperback_size_variant: true,
+                  paperback_volumes: true,
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  });
+  invariant(friend !== null);
+  const friendProps = {
+    ...friend,
+    quotes: friend.friend_quotes.map((q) => ({ quote: q.text, cite: q.source })),
+    documents: friend.documents.map((doc) => {
+      const firstEdition = doc.editions[0];
+      invariant(firstEdition !== undefined);
+      invariant(firstEdition.edition_impressions !== null);
```

Comment:
this invariant is failing for compilations, lydia barclay, etc., due to draft status pointed out above.